compiler/src/iree/compiler/Dialect/Encoding/IR/EncodingInterfaces.td

```diff
@@ -90,4 +94,38 @@ def IREEEncoding_EncodingLayoutAttrInterface :
   ];
 }
 
+//===----------------------------------------------------------------------===//
+// Type Interfaces
+//===----------------------------------------------------------------------===//
+
+def IREEEncoding_EncodingTypeInterface :
+  TypeInterface<"EncodingTypeInterface"> {
+  let cppNamespace = "::mlir::iree_compiler::IREE::Encoding";
+
+  let description = [{
+    Interface used to access/update tensor types with encodings.
+  }];
+
+
+  let methods = [
+    InterfaceMethod<
+      [{
+        Returns the tensor type with the encoding.
+      }],
+      /*retTy=*/"::mlir::Type",
```

Comment:
Why not `RankedTensorType`?

Reply:
See the other comment for the similar reason. It is not guaranteed that they are always `RankedTensorType`. We (and external folks) can define their own type system.